apollo-router/src/graphql.rs

```diff
@@ -20,6 +22,16 @@ pub use crate::request::Request;
 pub use crate::response::IncrementalResponse;
 pub use crate::response::Response;
 
+/// An asynchronous [`Stream`] of GraphQL [`Response`]s.
+///
+/// In some cases such as with `@defer`, a single HTTP response from the Router
+/// may contain multiple GraphQL responses that will be sent at different times
+/// (as more data becomes available).
+///
+/// We represent this in Rust as a stream,
+/// even if that stream happens to only contain one item.
+pub type ResponseStream = Pin<Box<dyn Stream<Item = Response> + Send>>;
```

Comment:
It looks like we are making the decision to move from requiring the stream to be `'static` to being pinned. In general, I approve. Just so I really understand: am I correct?